File: codes/src/leetcode/ListNode.java
package leetcode;

public class ListNode {
    int val;
    ListNode next;

    public ListNode(int x) {
        val = x;
    }

    public ListNode(int[] a) {
        if (a != null && a.length > 0) val = a[0];
        ListNode p = this;
        for (int i = 1; i < a.length; i++) {
            p.next = new ListNode(a[i]);
            p = p.next;
        }
    }

    public boolean equals(int[] a) {
        ListNode p = this;
        for (int i = 0; i < a.length && p != null; i++, p = p.next) {
            if (a[i] != p.val) return false;
            if (i == a.length - 1 && p.next == null) return true;
        }
        return false;
    }
}

File: codes/src/leetcode/ListNode.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace leetcode
{
    public class ListNode
    {
        public int val;
        public ListNode next;

        public ListNode(int x)
        {
            val = x;
        }

        public ListNode(IEnumerable<int> nums)
        {
            using (var it = nums.GetEnumerator())
            {
                var p = this;
                if (it.MoveNext()) p.val = it.Current;
                while (it.MoveNext())
                {
                    p.next = new ListNode(it.Current);
                    p = p.next;
                }
            }
        }

        public List<int> ToList()
        {
            var ret = new List<int>();
            for (ListNode p = this; p != null; p = p.next)
            {
                ret.Add(p.val);
            }
            return ret;
        }
    }

}
